src/test/fuzz/tx_pool.cpp

```diff
@@ -0,0 +1,230 @@
+    {
+        Assert(m_added.insert(tx).second);
+    }
+
+    void TransactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRemovalReason reason, uint64_t /* mempool_sequence */) override
+    {
+        Assert(m_removed.insert(tx).second);
+    }
+};
+
+FUZZ_TARGET_INIT(tx_pool_standard, initialize_tx_pool)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    const auto& node = g_setup->m_node;
+
+    {
+        const auto time_fuzzed = ConsumeTime(fuzzed_data_provider);
+        SetMockTime(time_fuzzed);
+    }
+    // All spendable outpoints
+    std::set<COutPoint> outpoints;
+    for (const auto& outpoint : g_outpoints_coinbase_init) {
+        Assert(outpoints.insert(outpoint).second);
+        if (outpoints.size() >= COINBASE_MATURITY) break;
+    }
+    // The sum of the values of all spendable outpoints
+    constexpr CAmount SUPPLY_TOTAL{COINBASE_MATURITY * 50 * COIN};
+
+    CTxMemPool tx_pool;
+
+    // Helper to query an amount
+    const CCoinsViewMemPool amount_view{WITH_LOCK(::cs_main, return &node.chainman->ActiveChainstate().CoinsTip()), tx_pool};
+    const auto GetAmount = [&](const COutPoint& outpoint) {
+        Coin c;
+        amount_view.GetCoin(outpoint, c);
+        Assert(!c.IsSpent());
+        return c.out.nValue;
+    };
+
+    while (fuzzed_data_provider.ConsumeBool()) {
+        {
+            // Total supply is the mempool fee + all spendable outpoints
+            CAmount supply_now{WITH_LOCK(tx_pool.cs, return tx_pool.GetTotalFee())};
+            for (const auto& op : outpoints) {
+                supply_now += GetAmount(op);
+            }
+            Assert(supply_now == SUPPLY_TOTAL);
+        }
+        if (fuzzed_data_provider.ConsumeBool()) {
+            const auto time_fuzzed = ConsumeTime(fuzzed_data_provider);
+            SetMockTime(time_fuzzed);
+        }
+        if (fuzzed_data_provider.ConsumeBool()) {
+            node.args->ForceSetArg("-limitancestorcount",
+                                   ToString(fuzzed_data_provider.ConsumeIntegralInRange<unsigned>(0, 200)));
+            node.args->ForceSetArg("-limitancestorsize",
+                                   ToString(fuzzed_data_provider.ConsumeIntegralInRange<unsigned>(0, 200)));
+            node.args->ForceSetArg("-limitdescendantcount",
+                                   ToString(fuzzed_data_provider.ConsumeIntegralInRange<unsigned>(0, 200)));
+            node.args->ForceSetArg("-limitdescendantsize",
+                                   ToString(fuzzed_data_provider.ConsumeIntegralInRange<unsigned>(0, 200)));
+            node.args->ForceSetArg("-maxmempool",
+                                   ToString(fuzzed_data_provider.ConsumeIntegralInRange<unsigned>(0, 200)));
+            node.args->ForceSetArg("-mempoolexpiry",
+                                   ToString(fuzzed_data_provider.ConsumeIntegralInRange<unsigned>(0, 200)));
```

Comment:
Question: I know it's a test, but does it make sense for any of these values to be 0? 😮 I'm also wondering why 200 is the max instead of something closer to the default values?

Reply:
* I want to set it small enough so that the fuzz engine can quickly hit the limit without having to produce megabytes of transactions first.
* Changed all maxima to be multiples of the default values, except for maxmempool, which is in [0,30]